emData/download.sh

```diff
@@ -161,6 +173,8 @@ then
   rm -f LUTs.tar.gz
 fi
 
+./generate_VMR.py -a -w LUTs/wires.dat
```

Comment:
I added protection against missing include files to the VMRouter_parameters.h that is generated by generate_VMR.py (and similar protections in files for the TC and MC) because some tables are not even produced when running the emulation with the reduced configuration (e.g., no tables for the disk VMRs). So the above line now looks like this:
```c++
  // TE Inner Memory 3
  ap_uint<1> tmpBendTable3n1[bendCutTableSize] =
#if __has_include("../emData/VMR/tables/VMSTE_L1PHID15n1_vmbendcut.tab")
#  include "../emData/VMR/tables/VMSTE_L1PHID15n1_vmbendcut.tab"
#else
  {};
#endif
```

The CI was failing because the CI configuration for the code quality check was still pointing to the old VMRouterTop.cc, which doesn't exist anymore. I just updated it, so it should pass now.

You're right though that we could just generate exactly the VMRs we need for a given configuration. However, I do like the idea of simply replacing the wiring and rerunning download.sh without modification in order to update all the top functions. That being said, we do still have to update this line in generate_VMR.py when switching to a different configuration:
https://github.com/cms-L1TK/firmware-hls/blob/bfffeda52e8bb0107c1fac48cb0c0ac98c10d7c6/emData/generate_VMR.py#L19-L20
Would it be difficult to have the script automatically get a list of all VMRs in the given wiring when using the --all option?

Reply:
The --all option already does that? It calls:
https://github.com/cms-L1TK/firmware-hls/blob/vmr_generate_top/emData/generate_VMR.py#L112
The defulat_vmr_list is just a list of random VMRs that I tend to test whenever I debug the VMR :-)

Reply:
My bad. The error I saw got scrambled up in my head.

The actual issue is when there are VMRs in the `default_vmr_list` that do not exist in the wiring. For example, when we switch to the reduced config, there are no disk VMRs and that raises an exception that I think is related to nonexistent memories:
```
Traceback (most recent call last):
  File "/home/hart/Research/CMS/tracklets/firmware-hls/emData/./generate_VMR.py", line 707, in <module>
    writeParameterFile(mem_dict)
  File "/home/hart/Research/CMS/tracklets/firmware-hls/emData/./generate_VMR.py", line 308, in writeParameterFile
    "  return " + str(len(mem_dict["IL_"+vmr])) + ";\n"
KeyError: 'IL_VMR_D1PHIA'
```

Obviously, we can just edit the `default_vmr_list` to get rid of the error, but I think when the --all option is used, it should just effectively ignore `default_vmr_list`, and maybe print a warning about VMRs in that list that are not in the given wiring. Would that be easy to do?

Reply:
Aaaah yes now I get it, that should be an easy fix.

Reply:
Should be done now